spec/unit/provider/oneview_server_profile_template_c7000_spec.rb

```diff
@@ -0,0 +1,150 @@
+      allow(OneviewSDK::ServerProfile).to receive(:find_by).and_return([server_profile])
+      provider.exists?
+      expect(provider.set_new_profile).to be_nil
+    end
+
+    it 'should be able to create a server profile with given name using the template' do
+      resource['data']['serverProfileName'] = 'New Server Profile'
+      test = resourcetype.new(@client, resource['data'])
+      allow(resourcetype).to receive(:find_by).with(anything, resource['data']).and_return([test])
+      allow(resourcetype).to receive(:find_by).with(anything, name: resource['data']['name']).and_return([test])
+      server_profile = OneviewSDK::ServerProfile.new(@client, name: 'New Server Profile')
+      allow(OneviewSDK::ServerProfile).to receive(:find_by).and_return([])
+      allow_any_instance_of(resourcetype).to receive(:new_profile).and_return(server_profile)
+      allow(server_profile).to receive(:create).and_return(server_profile)
+      provider.exists?
+      expect(provider.set_new_profile).to be
+    end
+
+    it 'should not create a server profile with given name using the template when already exists' do
+      resource['data']['serverProfileName'] = 'New Server Profile'
+      test = resourcetype.new(@client, resource['data'])
+      allow(resourcetype).to receive(:find_by).with(anything, resource['data']).and_return([test])
+      allow(resourcetype).to receive(:find_by).with(anything, name: resource['data']['name']).and_return([test])
+      server_profile = OneviewSDK::ServerProfile.new(@client, name: 'New Server Profile')
+      allow(OneviewSDK::ServerProfile).to receive(:find_by).and_return([server_profile])
+      provider.exists?
+      expect(provider.set_new_profile).to be_nil
+    end
+
+    it 'should be able to get a server profile template with a new configuration', if: api_version >= 300 do
+      resource['data']['queryParameters'] = {
+        'enclosureGroupUri'     => 'NameInterconn40GB',
+        'serverHardwareTypeUri' => 'SY 480 Gen9 1'
+      }
+      test = resourcetype.new(@client, resource['data'])
+      fake_server_profile = { 'name' => 'Fake profile template with a new configuration' }
+      allow(resourcetype).to receive(:find_by).and_return([test])
+      allow_any_instance_of(resourcetype).to receive(:get_transformation).and_return(fake_server_profile)
+      expect(provider.get_transformation).to be
+    end
+
+    it 'should be able to delete the resource' do
+      resource['data'] = { 'name' => 'SPT', 'uri' => '/rest/fake' }
+      test = resourcetype.new(@client, resource['data'])
+      allow(resourcetype).to receive(:find_by).with(anything, resource['data']).and_return([test])
+      allow(resourcetype).to receive(:find_by).with(anything, name: resource['data']['name']).and_return([test])
+      expect_any_instance_of(OneviewSDK::Client).to receive(:rest_delete).and_return(FakeResponse.new('uri' => '/rest/fake'))
```

Comment:
Should use instances of `resourcetype` instead of `OneviewSDK::Client` and receive `:delete` instead of `:rest_delete`